tests/pydis_core/test_api.py

```diff
@@ -17,18 +17,15 @@ def test_response_code_error_default_initialization(self):
         """Test the default initialization of `ResponseCodeError` without `text` or `json`"""
         error = site_api.ResponseCodeError(response=self.error_api_response)
 
-        self.assertIs(error.status, self.error_api_response.status)
-        self.assertEqual(error.response_json, {})
-        self.assertEqual(error.response_text, None)
-        self.assertIs(error.response, self.error_api_response)
+        assert error.status is self.error_api_response.status
```

Comment:
Do we want such a change ? the assertion methods offered by unittest do a lot more things than the native one. For instance, comparing lists is easier as it points out the difference between them, etc.

Reply:
I don't really mind either way, can revert this for now to get it merged